File: Legacy-code-Migrator-main/app/BankApp.java
package com.bank;

import com.bank.core.BankAccount;
import com.bank.core.Customer;
import com.bank.services.LoanCalculator;

/**
 * Main entry point for the Banking Application.
 * Demonstrates account operations, transfers, and loan calculations.
 */
public class BankApp {

    public static void main(String[] args) {
        System.out.println("+======================================+");
        System.out.println("|     WELCOME TO LEGACY BANK APP       |");
        System.out.println("+======================================+\n");

        // Create customers
        Customer alice = new Customer("Alice Johnson", "alice@email.com", "555-0101");
        Customer bob = new Customer("Bob Smith", "bob@email.com", "555-0202");

        // Open accounts
        BankAccount aliceSavings = alice.openAccount("SAVINGS", 5000.00);
        BankAccount aliceChecking = alice.openAccount("CHECKING", 2000.00);
        BankAccount bobSavings = bob.openAccount("SAVINGS", 3000.00);

        // Perform transactions
        aliceSavings.deposit(1500.00);
        aliceChecking.withdraw(500.00);
        aliceSavings.transferTo(bobSavings, 1000.00);

        // Edge cases
        aliceChecking.withdraw(99999.00); // Should fail - insufficient funds
        aliceSavings.deposit(-100);       // Should fail - negative amount

        // Print summaries
        alice.printCustomerSummary();
        bob.printCustomerSummary();

        // Loan calculation
        System.out.println("\n--- LOAN CALCULATOR ---");
        LoanCalculator homeLoan = new LoanCalculator(250000.00, 6.5, 360);
        System.out.printf("Home Loan Monthly Payment: $%.2f%n", homeLoan.calculateMonthlyPayment());
        System.out.printf("Total Interest: $%.2f%n", homeLoan.calculateTotalInterest());

        LoanCalculator carLoan = new LoanCalculator(30000.00, 4.5, 60);
        carLoan.printAmortizationSchedule();

        System.out.println("\nBanking application completed successfully.");
    }
}

File: Legacy-code-Migrator-main/migrated_output/app/bankapp.py
# Migrated from BankApp.java using AI Migration Agent
"""Banking Application migrated to Python.

This module demonstrates account operations, transfers, and loan calculations.
It preserves the original business logic from the Java version.
"""

from __future__ import annotations

# Assuming equivalent Python modules exist for the legacy Java classes.
# Adjust import paths as necessary for your project structure.
from bank.core import BankAccount, Customer
from bank.services import LoanCalculator


def main() -> None:
    """Entry point for the Banking Application.

    Replicates the behavior of the original Java `BankApp.main` method.
    """
    print("+======================================+")
    print("|     WELCOME TO LEGACY BANK APP       |")
    print("+======================================+\n")

    # Create customers
    alice = Customer(name="Alice Johnson", email="alice@email.com", phone="555-0101")
    bob = Customer(name="Bob Smith", email="bob@email.com", phone="555-0202")

    # Open accounts
    alice_savings: BankAccount = alice.open_account(account_type="SAVINGS", initial_balance=5000.00)
    alice_checking: BankAccount = alice.open_account(account_type="CHECKING", initial_balance=2000.00)
    bob_savings: BankAccount = bob.open_account(account_type="SAVINGS", initial_balance=3000.00)

    # Perform transactions
    alice_savings.deposit(1500.00)
    alice_checking.withdraw(500.00)
    alice_savings.transfer_to(bob_savings, 1000.00)

    # Edge cases (expected to raise exceptions or be handled internally)
    try:
        alice_checking.withdraw(99999.00)  # Should fail - insufficient funds
    except Exception as e:
        print(f"Withdrawal failed as expected: {e}")

    try:
        alice_savings.deposit(-100)  # Should fail - negative amount
    except Exception as e:
        print(f"Deposit failed as expected: {e}")

    # Print summaries
    alice.print_customer_summary()
    bob.print_customer_summary()

    # Loan calculation
    print("\n--- LOAN CALCULATOR ---")
    home_loan = LoanCalculator(principal=250_000.00, annual_rate_percent=6.5, term_months=360)
    print(f"Home Loan Monthly Payment: ${home_loan.calculate_monthly_payment():.2f}")
    print(f"Total Interest: ${home_loan.calculate_total_interest():.2f}")

    car_loan = LoanCalculator(principal=30_000.00, annual_rate_percent=4.5, term_months=60)
    car_loan.print_amortization_schedule()

    print("\nBanking application completed successfully.")


if __name__ == "__main__":
    main()
